armi/reactor/grids.py

```diff
@@ -1073,6 +1099,35 @@ def getRingPos(self, indices):
             pos = 7 * ring + j
         return (int(ring) + 1, int(pos) + 1)
 
+    def getMinimumRings(self, n):
+        """
+        Return the minimum number of rings needed to fit ``n`` objects.
+        """
+        return self._getMinimumRingsStatic(n, throughCenter=self._isThroughCenter())
+
+    @staticmethod
+    def _getMinimumRingsStatic(n, throughCenter):
+        # Keep private because this is really here to support old locations stuff that
+        # hasn't been removed yet.
+        numPositions = 0
+        for ring in itertools.count(1):
+            ringPositions = self._getPositionsInRing(ring, throughCenter)
+            numPositions += ringPositions
+            if numPositions >= n:
+                return ring
+
+    @staticmethod
+    def _getPositionsInRing(ring, throughCenter):
+        # Keep private because this is really here to support old locations stuff that
+        # hasn't been removed yet.
```

Comment:
not sure I understand that comment

Reply:
This is pretty weird and nuanced, so yeah probably needs a better comment. More context:
Cartesian grids need state (throughCenter) in order to reason about ring/pos. So for ring/pos to be part of the base Grid abstraction, we need to assume that _any_ Grid can only cough up ring/pos stuff with an instance method. HexGrid is sort of the special case, in that it _could_ do ring/pos stuff as a static method (notice that it punts the actual implementation to hexagon.py). There is no similar `rectangle.py` to hold the concept of ring/pos for the CartesianGrid case, and to do such a thing would be extra weird, because a throughCenter doesn't really mean anything to a simple rectangle; the grid adds the meaning to throughCenter.